### PR TITLE
pytorch: renamed from libtorch

### PR DESCRIPTION
**Pull request by singingwolfboy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Alternative to #110103. This renames `libtorch` to `pytorch`, and installs the Python module as well as the C++ libraries.

### CODE REVIEW

**Comment by singingwolfboy** [comment]
Seems like some of the CI builds are failing when trying to install `torchvision`, because `pytorch` doesn't have bottles built. Not sure why this is happening, or how to resolve it...

File: Formula/torchvision.rb [hunk 1]
```diff
@@ -59,9 +59,9 @@ def install
       %w[-fopenmp]
     end
     system ENV.cxx, "-std=c++14", "test.cpp", "-o", "test", *openmp_flags,
-                    "-I#{libtorch.opt_include}",
-                    "-I#{libtorch.opt_include}/torch/csrc/api/include",
-                    "-L#{libtorch.opt_lib}", "-ltorch", "-ltorch_cpu", "-lc10",
+                    "-I#{pytorch.opt_include}",
+                    "-I#{pytorch.opt_include}/torch/csrc/api/include",
+                    "-L#{pytorch.opt_lib}", "-ltorch", "-ltorch_cpu", "-lc10",
```
Comment (danielnachun) [inline Formula/torchvision.rb:64]: I think on Linux you'll also need to add `-Wl,-rpath,#{pytorch.opt_lib}` for this test to work.

Comment (singingwolfboy) [reply Formula/torchvision.rb:64]: The test is working with the existing `libtorch` formula, so I think it's fine

**Comment by singingwolfboy** [comment]
I'm seeing this error message on Linux:

```
The formula built, but is not symlinked into /home/linuxbrew/.linuxbrew
Error: The `brew link` step did not complete successfully
Could not symlink lib/python3.10/site-packages/typing_extensions-4.3.0.dist-info/INSTALLER
Target /home/linuxbrew/.linuxbrew/lib/python3.10/site-packages/typing_extensions-4.3.0.dist-info/INSTALLER
already exists. You may want to remove it:
  rm '/home/linuxbrew/.linuxbrew/lib/python3.10/site-packages/typing_extensions-4.3.0.dist-info/INSTALLER'

To force the link and overwrite all conflicting files:
  brew link --overwrite pytorch

To list all files that would be deleted:
  brew link --overwrite --dry-run pytorch
```

But I'm not sure why there would be a conflict on this file for Linux, but not on Mac. Does anyone have any clues why that might be?

I could create a venv just for this `typing_extensions` module, but that seems like overkill. I could make a new `python-typing-extensions` formula for Homebrew, and make this formula depend on that, like it does for `pyyaml`. Or if I can figure out why it's only a problem on Linux, I could resolve it that way. Any suggestions on the best way to proceed?

**Comment by singingwolfboy** [comment]
Is it OK to mark this `pytorch` formula as only working on ARM64 and Linux? (See #111718) I'm not sure how to express that in a formula: it's not precisely `depends_on arch: :arm64`, because it also works on Linux.

**Comment by singingwolfboy** [comment]
@carlocab I finally got the tests passing on this pull request, so I think it's ready for review and (hopefully!) merge. I had to disable this formula for Intel Mac, because `pytorch` does not build with LLVM 14 (see https://github.com/Homebrew/homebrew-core/pull/111718). Happy to answer other questions, if you have any!

**Comment by singingwolfboy** [comment]
There are so many pull requests in this repo that I don't think the older ones get much attention. I'm going to close this pull request and open a new one.

**Pull request closed without merging**